File: IT/Stringhe/Java/StringConcat.java
/* file StringConcat.java */

public class StringConcat {
	public static void main(String [] args) {
		String hello = "Ciao, ";
		String world = "mondo!";

		/* concatena le due stringhe, usando l'operatore "+" */

		String s1 = hello + world;
		System.out.println(s1);

		/* ...o il metodo String.concat() */

		String s2 = hello.concat(world);
		System.out.println(s2);

		if (s1.equals(s2)) {
			System.out.println("Le stringhe \"" + s1 + "\" e \"" + s2 + "\" sono uguali.");
		}

		int n = 1;
		String s3 = " è un numero positivo.";

		/* concatena le due stringhe: il numero 1 viene automaticamente convertito in una stringa */

		System.out.println(n + s3);
	}
}

File: IT/Stringhe/C++/stringConcat.cpp
/*
file stringConcat.cpp
compilare con: g++ -std=c++11 stringConcat.cpp
*/

#include <iostream>
#include <sstream>

using std::cout;
using std::endl;
using std::string;
using std::to_string;
using std::stringstream;

int main()
{
	string hello = "Ciao, ";
	string world = "mondo!";

	/* concatena le due stringhe, usando l'operatore "+" */

	string s1 = hello + world;

	cout << s1 << endl;

	/* ...o la funzione string::append(), che modifica la stringa originale */

	cout << hello.append(world) << endl;

	if (hello == s1) {
		cout << "La stringa iniziale \"" << hello << "\" e quella finale \"" << s1 << "\" sono uguali." << endl;
	}

	/* converte il numero 1 nella stringa '1' */

	int n = 1;
	string s2 = " è un numero positivo.";

	/* concatena le due stringhe, usando gli stream */

	stringstream stream;
	stream << n;

	cout << stream.str() + s2 << endl;

	/* ...o la nuova funzione to_string() */

	cout << to_string(n) +  s2 << endl;
}

